src/main/scala/com/normation/cfclerk/domain/HashAlgoConstraint.scala

```diff
@@ -72,11 +72,20 @@ object HashAlgoConstraint {
   /*
    * Actually do not hash the result
    */
-  object PLAIN  extends HashAlgoConstraint {
+  object PLAIN extends HashAlgoConstraint {
     override def hash(input:Array[Byte]) : String = new String(input, "UTF-8")
     override val prefix = "plain"
   }
 
+  object PreHashed extends HashAlgoConstraint {
+    override def hash(input:Array[Byte]) : String = new String(input, "UTF-8")
+    override val prefix = "pre-hashed"
+  }
+
+  object SCRIPT extends HashAlgoConstraint {
+    override def hash(input:Array[Byte]) : String = new String(input, "UTF-8")
+    override val prefix = "script"
```

Comment:
doesn't it have other impact in term of code ?
what is it expected to do when it's a script ?